### ISSUE TITLE
Do not scroll to previously active tab when closing active tab

### ISSUE THREAD

**Issue by Compizfox** [mbnuqw/sidebery#260]
My apologies about the cryptic title. My request refers to the following situation:

- I have more tabs open in a panel than fit in the window (so there is a scroll bar)
- I have "after closing current active tab activate" set to "previously active tab"

When I close the currently active tab and the previously active tab is on the opposite side of the panel, the previously active tab is not only activated, but the panel also scrolls to that tab.

To me, the latter behaviour is quite annoying because it frequently causes me to misclick when I want to activate/close another tab right after closing the active tab.

It would already be a big improvement if the scroll was animated (it does not currently appear to be), but actually I would welcome an option that turns off the scrolling behaviour completely, so the tab is only activated but not scrolled to.

**Comment by mbnuqw** [comment]
Ok, I can add option to completely disable auto-scroll.

**Comment by deviant** [comment]
Hi, is this a thing yet?

I often find myself in a position where I have a swath of unloaded tabs, and will go through a few of them while closing them. With the current default behaviour, whenever I close one of the tabs I just loaded, I get bounced back to my most recently active tab (often right at the end of the panel), and the panel scrolls to make this visible, taking me out of the region I was closing tabs in. I then have to scroll all the way back up, only for the same thing to happen again a few seconds later. This can be quite annoying.

I would find it useful to be able to keep the scroll position in one place while closing these tabs, but for it to reset to show the currently open tab once I'm done with closing. This could work in the same way that the "Automatically switch panel on activating tab of another panel" option functions when set to "if the mouse left the sidebar". i.e., retain the auto-scrolling logic, but wait for the user to be done interacting with Sidebery first. I've been using that option for quite some time and find it increases the usability by a lot (and perhaps should be the default?)

Thank you for your consideration!

**Comment by Hoernchen** [comment]
This is kind of the one thing that makes sideberry currently unusable for me, always almost random jumps 1000 tabs up or to another tab panel..

**Comment by the-nelsonator** [comment]
@Compizfox @deviant @Hoernchen do "next tab" or "previous tab" options for "After closing current active tab activate" work for you? Avoids scrolling after closing active tab in _most_ cases. If not, what behavior are you looking for?

I also notice there's an "off" option but I'm not sure how this is any different from "next tab"

**Comment by deviant** [comment]
> do "next tab" or "previous tab" options for "After closing current active tab activate" work for you? Avoids scrolling after closing active tab in _most_ cases. If not, what behavior are you looking for?

Not really, since I do want it to return to the most recent tab I previously had activated. This is purely about scrolling the Sidebery panel. Next tab/previous tab would alter browser behaviour, and then require me to scroll around manually to find where I had previously been, kind of defeating the point.

**Comment by Hoernchen** [comment]
The meaning of "off" for "After closing current active tab activate"  is explained at https://github.com/mbnuqw/sidebery/issues/1804#issuecomment-2372489092 , IMHO a big documentation issue because off is off for me and not "default", the expected behavior was _no scrolling_ to be honest, and there is no other setting that _kinda_ looks like what I am looking for. Yes, I have now learned that activation and scrolling are two different things, and that setting probably works as intended, but closing the bottom most tab or a bunch of tabs in the middle and then ending up 100 tabs above or in another panel is very frustrating.

> Avoids scrolling after closing active tab in most cases. 

And that's the thing.. it feels erratic. I honestly cant tell you why it randomly jumps around at times, and appears to not scroll at other times. I guess one half of the problem is 1) not deliberately activating tabs "nearby" when closing first 2) pinned tabs, i.e. pinned whatsapp -> middle click on a link I got, the new tab ends up all the way at the top of the tab list, afterwards I press ctr-t for a new tab that shows up at the bottom, some more usage at the bottom continues, I close tabs at the bottom, BAM, I'm suddenly up all the way again. Or just scrolling up and down and noticing a cluster of inactive tabs somewhere that needs to be closed without strategically activating one nearby first.


I suppose a small "temporarily disable all scrolling" button that can be added next to the trash can and bookmarks buttons at the bottom would help?

**Comment by Hoernchen** [comment]
I gave that a shot: https://github.com/mbnuqw/sidebery/commit/9dce4cfe2e9bd3e1bfb178e7b1fcd92584dcc7b0

**Comment by the-nelsonator** [comment]
@Hoernchen nice! you should open a PR!

**Comment by deviant** [comment]
I took a crack at this myself, and implemented the behaviour I described wanting upthread. I don't write a lot of JavaScript these days, and this was an unfamiliar codebase to me, but I found it a lot easier to manage than I had expected I might!

The "if the mouse left the sidebar" option works very nicely IMO and is a good middle ground between enabled and disabled, since it still scrolls the active tab into view, but doesn't get in the way of the user whilst they're closing tabs.

**Comment by llc0930** [comment]
@mbnuqw Currently, the behavior of "settings.scroll_panel_after_switching_tab" is inconsistent with the description in the English translation.

Depending on the value of "Settings.state.scrollPanelAfterSwitchingTab" in the "settings.scroll_panel_after_switching_tab" setting, this setting affects the scrolling behavior when switching tabs in the current panel using the tab flip feature ("settings.tabs_second_click_act_prev"), as well as the scrolling behavior after clicking a search result.

In addition, when "settings.activate_after_closing" is set to "prev_act", the "settings.scroll_panel_after_switching_tab" setting will affect the final display position of the two panels after closing a tab, depending on the setting of "settings.switch_panel_after_switching_tab" and the current display position of the two tabs' respective panels. Of course, if the previous tab is also in the same panel, it will also be affected... which is obviously inconsistent with the description in the English translation.
